File: FlankSpeedLegacyDemo.Web/Models/ApprovalStep.cs
using System.ComponentModel.DataAnnotations;

namespace FlankSpeedLegacyDemo.Web.Models;

// Used primarily by CWA module for multi-step approval workflow
public class ApprovalStep
{
    public int Id { get; set; }
    public int WorkItemId { get; set; }
    public WorkItem? WorkItem { get; set; }

    // e.g., "Supervisor Review", "Director Approval"
    [Display(Name = "Step")]
    public string StepName { get; set; } = string.Empty;

    public int StepOrder { get; set; }

    public int? ApproverUserId { get; set; }
    [Display(Name = "Approver")]
    public User? ApproverUser { get; set; }

    // "Pending", "Approved", "Rejected", "Returned"
    public string Decision { get; set; } = "Pending";

    [Display(Name = "Decision Date")]
    public DateTime? DecisionDate { get; set; }

    public string? Remarks { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: power-apps-migration/src/models/ApprovalStep.ts
/**
 * Migrated from: FlankSpeedLegacyDemo.Web/Models/ApprovalStep.cs
 * Source: Multi-step approval workflow for CWA module
 * Target: Azure SQL [dbo].[ApprovalSteps] via Power Platform SQL connector
 *
 * Migration note: In the Power Apps target, approval routing logic is handled by
 * Power Automate flows triggered on WorkItem creation. The approval UI displays
 * step status and allows approvers to take action via the Code App interface.
 */
import { User } from "./User";

export interface ApprovalStep {
  id: number;
  workItemId: number;
  /** e.g., "Supervisor Review", "Director Approval", "Security Clearance Verification" */
  stepName: string;
  stepOrder: number;
  approverUserId: number | null;
  approverUser?: User;
  /** "Pending", "Approved", "Rejected", "Returned" */
  decision: ApprovalDecision;
  decisionDate: string | null;
  remarks: string | null;
  createdAt: string;
}

export type ApprovalDecision = "Pending" | "Approved" | "Rejected" | "Returned";

export const APPROVAL_DECISIONS: ApprovalDecision[] = [
  "Pending",
  "Approved",
  "Rejected",
  "Returned",
];
